### PR TITLE
jc 1.10.6 (new formula)

### PR DESCRIPTION
**Pull request by kellyjonbrazil**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by kellyjonbrazil** [comment]
Hmm, not sure what to do since my app is a compiled python binary.

```
$ brew audit --new-formula jc
jc:
  * C: 2: col 3: Description is too long. "name: desc" should be less than 80 characters. Length is calculated as jc + desc. (currently 116)
  * C: 2: col 119: Line is too long. [121/118]
  * C: 4: col 3: https://jc-packages.s3-us-west-1.amazonaws.com/bin/jc-1.10.5-darwin.tar.gz looks like a binary package, not a source archive; homebrew/core is source-only.
  * C: 13: col 119: Line is too long. [130/118]
  * Stable: version 1.10.5 is redundant with version scanned from URL
  * Files were found with references to the Homebrew shims directory.
    The offending files are:
      bin/jc
Error: 6 problems in 1 formula detected
```

File: Formula/jc.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Jc < Formula
+  desc "This tool serializes the output of popular gnu linux command-line tools and file types to structured JSON output"
```
Comment (jonchang) [inline Formula/jc.rb:2]: `desc` should be shorter

File: Formula/jc.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  homepage "https://github.com/kellyjonbrazil/jc"
+  url "https://jc-packages.s3-us-west-1.amazonaws.com/bin/jc-1.10.5-darwin.tar.gz"
```
Comment (jonchang) [inline Formula/jc.rb:4]: Formulae should build from source in homebrew-core: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

For an example of a python app formula, run `brew cat black`

File: Formula/jc.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  sha256 "b97f2f0fb612b75adbbcfd5956bc406e4892a8cffa809bea942e8eae4fa09a99"
+
+  def install
+    bin.install "jc"
+  end
+
+  test do
+    assert_equal '[{"header1": "data1", "header ": "data2"}]', pipe_output("#{bin}/jc --csv", 'header1, header "\n" data1, data2')
+  end
+end
```
Comment (jonchang) [inline Formula/jc.rb:15]: Files should end with a newline.

**Review by jonchang — CHANGES_REQUESTED** [review]
Looks like a good start! Please run `brew audit --new-formula --strict jc` locally and ensure that it passes. In addition, please squash your commits and amend the commit message to read `jc 1.10.5 (new formula)` per our guidelines: https://docs.brew.sh/Formula-Cookbook#commit

**Comment by kellyjonbrazil** [comment]
I think I'm pretty close now (except for the commit squashing, which I'll figure out).

I used the black package method and it installed fine, but my app errors out with this message when run.  Does this mean I need to go to all of my required packages and include their required packages, and so on?

```
Traceback (most recent call last):
  File "./jc", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3259, in <module>
    def _initialize_master_working_set():
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/Cellar/jc/1.10.5/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'ruamel.yaml.clib>=0.1.2' distribution was not found and is required by ruamel.yaml
```

**Comment by kellyjonbrazil** [comment]
This seems to work now. I'm not sure if I squashed and renamed the commit correctly, though.

**Comment by jonchang** [comment]
You've got a merge commit in there which our CI system can struggle with. I'll repair the branch in a second. In the future I recommend opening pull requests from branches, rather than the `master` branch of your fork.

**Comment by kellyjonbrazil** [comment]
Sorry about that - I'll definitely do that next time. I think we are good to go. Just fixed the test so it's passing now.

**Comment by kellyjonbrazil** [comment]
I think these errors are benign, but let me know if you think I need to make any adjustments.  The test fixtures have many file listings, so it makes sense that there might be a reference to the Homebrew shims directory. The `jc` tests are just parsing the text, so there is no real dependency.

```
$ brew audit --strict jc
jc:
  * C: 45: col 1: 1 trailing blank lines detected.
  * Files were found with references to the Homebrew shims directory.
    The offending files are:
      libexec/lib/python3.8/site-packages/tests/fixtures/osx-10.11.6/du.out
      libexec/lib/python3.8/site-packages/tests/fixtures/osx-10.11.6/du.json
      libexec/lib/python3.8/site-packages/tests/fixtures/osx-10.14.6/du.out
      libexec/lib/python3.8/site-packages/tests/fixtures/osx-10.14.6/du.json
Error: 2 problems in 1 formula detected
```

**Comment by kellyjonbrazil** [comment]
All tests are passing now.

**Comment by kellyjonbrazil** [comment]
I'm not sure how to squash the commits without causing a merge issue. Shall I close this PR, create a new local branch, and submit a new PR from there with a single commit so it will be clean?

**Comment by SMillerDev** [comment]
Rebase your branch, squash the commits and (force) push the resulting commit.

**Comment by kellyjonbrazil** [comment]
I think I've squashed as much as I can since there is a weird merge in my branch I don't know how to deal with.

**Comment by kellyjonbrazil** [comment]
Ok, I think we are good now!

**Comment by SMillerDev** [comment]
Thanks @kellyjonbrazil  Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by jconley** [comment]
fyi the catalina bottle is returning 404

**Comment by Bo98** [comment]
Thanks. I’ll check up on why that happened.

**Comment by Bo98** [comment]
Should be fixed now.

**Pull request closed without merging**